server/server.py

```diff
@@ -6,21 +6,37 @@
 import logging
 import sys
 import toml
+import structlog
+
 
 dictConfig({
     'version': 1,
+    'disable_existing_loggers': False,
     'formatters': {
         'default': {
-            'format': '%(asctime)s - %(levelname)s @ ' +
-                      '%(module)s-%(funcName)s : %(message)s'
+            'format': '{Time: %(asctime)s, Level: [%(levelname)s], ' +
+                      'module: %(module)s, function: %(funcName)s():%(lineno)s, message: %(message)s}',
+            "()": structlog.stdlib.ProcessorFormatter,
+            "processor": structlog.dev.ConsoleRenderer(colors=True),
+            'datefmt': '%Y-%m-%d %H:%M:%S',
+        },
+        "colored": {
+            'format': '{Time: %(asctime)s, Level: [%(levelname)s], ' +
+            'module: %(module)s, function: %(funcName)s():%(lineno)s, message: %(message)s}',
+            "()": structlog.stdlib.ProcessorFormatter,
+            "processor": structlog.dev.ConsoleRenderer(colors=True),
+            'datefmt': '%Y-%m-%d %H:%M:%S',
         }},
-    'handlers': {'wsgi': {
-        'class': 'logging.StreamHandler',
-        'stream': 'ext://flask.logging.wsgi_errors_stream',
-        'formatter': 'default'
-    }},
+    'handlers': {
+        'wsgi': {
+            'class': 'logging.StreamHandler',
+            'stream': 'ext://flask.logging.wsgi_errors_stream',
+            'formatter': 'colored'
+        }
+    },
     'root': {
         'level': 'INFO',
+        'propogate': True,
```

Comment:
> The propagate entry is set to 1 to indicate that messages must propagate to handlers higher up the logger hierarchy from this logger, or 0 to indicate that messages are not propagated to handlers up the hierarchy.

From what it seems from the docs, yes. But it needs some testing.

Reply:
`SystemExit` is a `BaseException` and doesn't print anything if not handled. I have noticed that adding a `raise SystemExit` will restart the server every time it runs into that piece which stops it from ever printing out anything meaningful.

> This exception is raised by the sys.exit() function. It inherits from BaseException instead of Exception so that it is not accidentally caught by code that catches Exception. This allows the exception to properly propagate up and cause the interpreter to exit. When it is not handled, the Python interpreter exits; no stack traceback is printed.

**For uncaught exceptions**
I do know that when I just insert a `raise Exception("uncaught exception")` in server.py like so
```
@app.route('/slack/commands', methods=['POST'])
def handle_commands():
    """Handle rocket slash commands."""
    txt = request.form['text']
    uid = request.form['user_id']
    raise Exception("UNCAUGHT EXCEPTION")
    return core.handle_app_command(txt, uid)
```

it prints out 
```
{Time: 2019-01-26 16:25:10, Level: [INFO], module: user, function: __init__():36, message: Initializing UserCommand instance}
{Time: 2019-01-26 16:25:14, Level: [ERROR], module: app, function: log_exception():1761, message: Exception on /slack/commands [POST] exc_info=(<class 'Exception'>, Exception('UNCAUGHT EXCEPTION'), <traceback object at 0x7f195617b888>)}
```